students/kegan/session04/which_languages.py

```diff
@@ -0,0 +1,44 @@
+"""
+Kathryn Egan
+
+In the class repo, in:
+
+Examples/Session01/students.txt
+
+You will find the list I generated in the first class
+of all the students in the class, and what programming
+languages they have used in the past.
+
+Write a little script that reads that file, and generates
+a list of all the languages that have been used.
+
+Extra credit: keep track of how many students specified each language.
+"""
+
+
+def main():
+    path_to_file = '/Users/kathryn/Desktop/Programming in Python/IntroPython-2017/examples/Session01/students.txt'
+
+    languages = {}
+
+    with open(path_to_file, 'r') as f:
+        f.readline()
+        for line in f.readlines():
```

Comment:
`for line in f:`

does the same thing, but without loading it all into memory first.